requirements.txt

```diff
@@ -37,3 +37,7 @@ wandb>=0.16.4
 xdg-base-dirs>=6.0.1
 psutil>=6.0.0
 huggingface_hub[hf_transfer]>=0.1.8
+haystack-ai>=2.8
+milvus_haystack>=0.0.11
+docling-core[chunking]>=2.10.0
+sentence-transformers>=3.0.0
```

Comment:
Do we need this dependency?

Reply:
`sentence-transformers` package is needed if we use the `SentenceTransformersDocumentEmbedder` (at ingestion time) and `SentenceTransformersTextEmbedder` (at inference time). They are compatible with the granite embedding model and also with most of the models available on HF.
Anyway, I see the point and I'm wondering if we need to make this step configurable in some way. Assuming that we're not serving the embedding endpoint (`ilab model serve --model-path ....ibm-granite/granite-embedding-125m-english`), the alternative is what Haystack calls the [on-premise-embedders](https://docs.haystack.deepset.ai/docs/choosing-the-right-embedder#on-premise-embedders). 
What other embedding component could we use?

Reply:
Thank you for explaining, @dmartinol .  This clarifies a lot.   I think you are right that SentenceTransformersTextEmbedder is the best option.  I will see what I can do about getting this approved.